### ISSUE TITLE
Open source OpenCL Static C++ Kernel Language Extension

### ISSUE THREAD

**Issue by oscarbg** [HSA-Libraries/Bolt#165]
Hi,
sorry as for sure that's not the best place to post but I remember seeing at APU13 a presentation saying AMD would open source his "OpenCL Static C++ Kernel Language Extension" so all OpenCL implementations could take advantage of it.. I think it was in Bolt presentation so asking here as I also remember it's was said either Q1 2014 or H1 2014 so must be coming soon right?
Also Bolt takes advantage of it right now in OCL path right?

**Comment by bensander** [comment]
The c++ static kernel language is an open specification but has only been implemented by amd.  The bolt implementation on c++amp currently provides portability across several vendors on windows , and there is an open source project which will bring c++ amp support to Linux through SPIR and HSAIL in the near future.

**Comment by oscarbg** [comment]
Yes but I'm refering to what you "seem" to have said in  http://www.slideshare.net/DevCentralAMD/pt-4059-bensander "PT-4059, Bolt: A C++ Template Library for Heterogeneous Computing, by Ben Sander"
in both slide 12 you say:
"Provide translator tool for C++ static kernel language"
and you dedicate full slide 14 saying:
AMD C++ Static Kernel Language (aka “OpenCL-C++ Kernel Language
"New translator tool is designed to bring these benefits to any OpenCL™ Implementation"
and finally
"Translator expected to be available in Q1-2014"
so just asking if all is going acording to plan and will soon see this translator..
would be awesome!

**Comment by bensander** [comment]
Unfortunately the translator tool has been put on hold.  
We are evaluating the recently-released specification for "SYCL", which provides many of the benefits of the C++ kernel language and additionally provides the advantage of a robust single-source C++ development language.
